core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -184,4 +190,27 @@ private void failIfOverShardCountLimit(ClusterService clusterService, int shardC
         }
     }
 
+    private void resolveIndexBoost(SearchRequest searchRequest, ClusterState clusterState) {
+        if (searchRequest.source() == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> indexBoost = searchRequest.source().indexBoost();
+        if (indexBoost == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> concreteIndexBoost = new ArrayList<>();
+        for (Tuple<String, Float> ib : indexBoost) {
+            String[] concreteIndices =
+                indexNameExpressionResolver.concreteIndexNames(clusterState, IndicesOptions.lenientExpandOpen(), ib.v1());
```

Comment:
Done.
But I'm wondering if it makes this change breaking. Specifying index that doesn't exist used to be no op. Now returns error.